### DISCUSSION TITLE
How to use the existing  PNP Powershell commands with Sites.Selected Permission in Azure AD

### DISCUSSION THREAD

**Discussion by in8awaka** [pnp/powershell#1480] [General]
Hi All,

I am using ClientID and ClientSecret of SharePoint Add-in to connect with PNP.Powershell but now i am getting  warning for Legacy Authentication (because of secret) , 

![tempsnip](https://user-images.githubusercontent.com/49389070/147643888-8bad3b92-a29f-46fc-8bb7-bfec6d4763ed.png)

So to resolve this we are moving towards certificates so that it will not get stop working one day and for resolving when

I am trying to Read / Write data in SP site using PNP.PowerShell with and Azure AD App having Sites.Selected Permission. I generated the token using client id and certificate and get the data using Graph API (Invoke-WebRequest) method 

When i use the same token in PNP.Powershell using 
Connect-PnPOnline -Url $siteURL -AccessToken $accessToken

it connected but not giving any result in Get-PnpList it gives me 401 . 

so please update on the future releases if it is planned for executing the the Graph API Sites.Selected permission with Connect-PnpOnline., if yes please update when this will get released. 

Also please update when this secret things will get stopped.

![image](https://user-images.githubusercontent.com/49389070/147533693-d3096d82-e9e3-4ac9-834a-c447e32e9024.png)

**Comment by gautamdsheth** [comment]
Closing it as issue, moving it to discussion

**Comment by LeonArmston** [comment]
Hi

I raised this last year in Issue https://github.com/pnp/powershell/issues/383 and tried to exact same thing with Get-PnPList and it didnt work. I also blogged about it ([link](https://www.leonarmston.com/2021/03/testing-out-the-new-microsoft-graph-sharepoint-specific-site-collection-app-permissions-with-pnp-powershell/))

Erwin said
> This is unfortunately nothing we can fix. It's a limitation of SharePoint: the native SharePoint APIs (CSOM, SPO REST) do not handle the site specific permissions. Only calls made to the Graph work with this. As most of the cmdlets do not use the Graph but either CSOM or SPO REST for now this permission is relatively limited in it's use.

I think it is on the future roadmap to move PnP.PowerShell to predominately use Graph instead of CSOM but this is likely to be a long time as Graph is nohere near full feature parity for SharePoint compared to CSOM.

Thanks

Leon
